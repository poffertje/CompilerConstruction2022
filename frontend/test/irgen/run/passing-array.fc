
extern int printf(char[] format, ...);

void print_nth(int[] a, int idx) {
    printf("Item %d: %d\n", idx, a[idx]);
}

int main() {
    int[16] a;
    for (int i = 0 to 16)
        a[i] = i * 2 + 1;
    print_nth(a, 3);
    print_nth(a, 7);
    return 0;
}
