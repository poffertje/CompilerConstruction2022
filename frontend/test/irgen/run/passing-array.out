Item 3: 7
Item 7: 15
